src/main/java/prestudy/team4/board/domain/UserEntity.java

```diff
@@ -0,0 +1,45 @@
+package prestudy.team4.board.domain;
+
+import jakarta.persistence.*;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class UserEntity{
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "user_id")
+    private Integer userId;
+
+    @Column(nullable = false, unique = true)
+    private String username; // userid를 고유값으로 설정
+
+    @Column(nullable = false)
+    private String password;
+
+    @Column(unique = true, nullable = false)
+    private String email;
+
+    @Column(nullable = false)
+    private String nickname;
+
+    @Column(nullable = false)
+    private String name;
+
+    @Column(nullable = false)
+    private String role;
```

Comment:
role의 후보가 "ROLE_USER", "ROLE_ADMIN" 정도라면..
public enum Role {
    ROLE_USER, ROLE_ADMIN
} 으로 enum(열거형)을 만들어주어서 사용하는 것이 더 편리해요. (개발자에 따라 ADMIN으로 쓰려는 사람도 있을 것이고, 나중에 생긴 오타도 컴파일 단계에서 잡기 힘들어요.)

```suggestion
    @Enumerated(EnumType.String) @Column(nullable = false)
    private Role role;
```